### ISSUE TITLE
[BUG] support of dark mode in personal note

### ISSUE THREAD

**Issue by Gwang-Jin** [bibletime/bibletime#330]
**Describe the bug**
A clear and concise description of what the bug is. Please only describe one issue per bug report.

Hello,

When I use dark mode of macOS, I found that some text of "personal note" is still back and some text is white. 
I uploaded my screenshot.

**To Reproduce**

**Expected behavior**

**Screenshots**
<img width="1280" alt="screenshot" src="https://user-images.githubusercontent.com/36105353/126590698-6d62a662-ad0a-4fb6-bc75-26279b886ec8.png">

**System information (please complete the following information):**
 - OS and OS version: macOS Mojave 10.14.6
 - BibleTime Version : 3.0.0
 - How BibleTime was installed [e.g. from system packages, Snap, Flatpak, installer, compiled from source code]
 - Versions of modules related to the crash (if known)
 - (If compiled from source) Sword Version
 - (If compiled from source) Qt Version

**Additional context**
Add any other context about the problem here, e.g. including output of `bibletime --debug` and/or backtraces in case of crashes.

**Comment by gholmlund** [comment]
In the Personal commentary, were you using plain text or html?

**Comment by Gwang-Jin** [comment]
I am using html. Some html text is OK. But some html text is black.

**Comment by gholmlund** [comment]
I will try to reproduce this in the next couple of days.

**Comment by gholmlund** [comment]
I have reproduced this on MacOs and linux when using html in Personal commentary.

**Comment by gholmlund** [comment]
This bug happened because in normal (white) mode, black text is added to a Personal commentary entry in html mode.  This means the stored html contains a color tag for the black text. When this is viewed in dark mode the black text does not show on a black background.
I added code to check if we are in dark mode and then convert black text to white text for display. This does not change anything that is saved to disk.

**Comment by jaakristioja** [comment]
I don't think this is the best way to solve this. I see a couple of problems with the approach now introduced to the master and stable-3.0 branches:

- The change made feels more like a new feature than a bugfix. Hence I don't think it should make its way to stable-3.0 because it might hinder packagers of some distros not to switch to 3.0.1 completely, but cherry-pick patches on top of 3.0.0.
- I don't think all users would like to see BibleTime unconditionally change their chosen colors like that. If we provide this feature, we should also give the user the chance to toggle it. This would require changes to the UI, additional translatable strings etc which might not be proper to introduce in 3.0.1.
- If in dark mode we replace dark colors with lighter ones, should we also replace light colors with darker ones in non-dark modes? Or whenever they are too close to the background color? I think this might need some more thought. I don't currently see any easy solutions for providing sufficient visual contrast in this case where the user has (perhaps explicitly) specified colors to the commentary, but where the background color might change depending on the current theme or mode used. For any colors saved together with the commentary texts we have no way of knowing whether the user explicitly selected those colors or for what reason he selected them (or didn't explicitly select a different color).

Some more raw thoughts:
- If we keep this feature, it should be amended so that users have the option to toggle it.
- We may instead (or in addition) provide the optional feature of disabling saved colors for the display of commentaries, so that the default colors would be used.
- We may disable colors in editing (but in this case how should we handle texts which are already saved with colors).
- We may disable colors in editing by default, e.g. color info is not added unless the user explicitly changes the color of some text.
- Ideally, instead of allowing the users to specify colors for the commentary texts, it might make more sense to provide a set of "styles" for editing, where the color (and perhaps other properties) for specific styles might be specified elsewhere (e.g. by the theme/mode with the option to override the defaults using the settings dialog). However, implementing something like this is a lot of work and does not seem very realistic.

@gholmlund I propose we revert the change and drop the 3.0.1 milestone. I'm not sure whether this is "fixable" at all. What do you think?

**Comment by gholmlund** [comment]
@jaakristioja  You are right that this is not an easy problem to fix correctly. Butd as it is without this fix a user cannot use dark mode and the personal commentary. 

The other way I considered is to not have a dark mode for the personal commentary. Always show it in white background. 

I am Ok with reverting if you feel that is the best action.

**Comment by jaakristioja** [comment]
Reverted for now. The current workaround for users is to edit and change the color of the text.
